plugin/src/main/java/com/denizenscript/denizen/objects/WorldTag.java

```diff
@@ -1482,14 +1524,44 @@ public void adjust(Mechanism mechanism) {
                 NMSHandler.worldHelper.wakeUpAllPlayers(world);
             }
             // minor change: prior to 1.18, hasStorm/isRaining was not checked
-            if (getGameRuleOrDefault(GameRule.DO_WEATHER_CYCLE) && world.hasStorm()) {
+            if (world.getGameRuleValue(GameRule.DO_WEATHER_CYCLE) == null) {
+                if (world.getGameRuleDefault(GameRule.DO_WEATHER_CYCLE) == null) {
+                    throw new IllegalStateException("World " + world.getName() + " contains no GameRule " + GameRule.DO_WEATHER_CYCLE.getName());
+                }
+            }
+            if (world.hasStorm()) {
                 NMSHandler.worldHelper.clearWeather(world);
             }
```

Comment:
Pretty sure this is a breaking change because the actual value of the gamerule isn't checked anymore - but is there any reason this was changed/can't just use the method anymore?